### PR TITLE
Prevent DDBox popup closing after window scrolling for iPad in desktop mode (T845484)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/drop_down_box.js [hunk 1]
```diff
@@ -265,6 +268,16 @@ const DropDownBox = DropDownEditor.inherit({
         return this.callBase();
     },
 
+    _preventCloseHandler: function() {
+        const isInputFocused = !!this._popup.$content().find('input:focus').length;
```
Comment (DokaRus) [inline js/ui/drop_down_box.js]:
```js
import { getActiveElement } from '../core/dom_adapter';
<...>
_isNestedElementActive: function() {
  const activeElement = getActiveElement();
  return activeElement && this._popup.$content().get(0).contains(activeElement);
}

_shouldCloseOnTargetScroll: function() {
  return realDevice.deviceType === 'desktop' && this._canShowVirtualKeyboard() && this._isNestedElementActive();
}

```

**Review by DokaRus — DISMISSED** [review]

File: js/ui/drop_down_box.js [hunk 2]
```diff
@@ -213,6 +217,25 @@ const DropDownBox = DropDownEditor.inherit({
         return this.callBase();
     },
 
+    ///#DEBUG
```
Comment (San4es) [inline js/ui/drop_down_box.js]: My suggestion is to export `realDevice` object from the current module:

```
///#DEBUG
module.exports.realDevice = realDevice;
```
and use it directly in the test.

So, we can avoid adding a new method to the component.

File: testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js [hunk 3]
```diff
@@ -502,6 +506,39 @@ QUnit.module('popup options', moduleConfig, () => {
...
+                assert.expect(0);
+                return;
+            }
+
+            const $content = $('<input type="text" />');
+            const instance = new DropDownBox($('#dropDownBox'), {
+                contentTemplate: () => $content
+            });
+
+            instance.setRealDevice({ mac: isMac });
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js]: We should restore an original real device object at the test end.

**Review by San4es — APPROVED** [review]
LGTM

**Review by DokaRus — APPROVED** [review]

**Pull request merged**